src/main/java/com/pusher/rest/util/Crypto.java

```diff
@@ -0,0 +1,105 @@
+package com.pusher.rest.util;
+
+import com.google.gson.Gson;
+import org.abstractj.kalium.crypto.Random;
+import org.abstractj.kalium.crypto.SecretBox;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+/**
+ * Crypto is a class containing all the utility methods needed to do End to End Encryption for End to End Encrypted Channels.
```

Comment:
I feel like "End to End" is repeated unnecessarily. How about "End to End Encryption for Encrypted Channels"?